File: src/ruby/problem002.rb
start = Time.now.to_f

sum = 0;
prev = 1;
n = 2;
while (n <= 4000000)
  if (n%2 == 0)
    sum += n;
  end
  tmp = n;
  n = prev + n;
  prev = tmp;
end

puts sum
puts "#{(Time.now.to_f - start) * 1000} ms"

File: src/java/euler/Problem002.java
package euler;

/**
 *
 * @author Pascal
 */
public class Problem002 {

    public static void main(String[] args) {
        long start = System.currentTimeMillis();

        int sum = 0;
        int prev = 1;
        int n = 2;
        while (n <= 4000000) {
            if (n%2 == 0) {
                sum += n;
            }
            int tmp = n;
            n = prev + n;
            prev = tmp;
        }

        System.out.println((System.currentTimeMillis() - start) + " ms");        
        System.out.println(sum);
    }
}
